pkg/app/middleware_test.go

```diff
@@ -0,0 +1,59 @@
+package app
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/pkg/thread"
+	"github.com/sirupsen/logrus"
+)
+
+func TestMiddleWare(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Middleware test")
+}
+
+var _ = Describe("Middleware test", func() {
+
+	l := logrus.New()
+	l.SetOutput(ioutil.Discard)
+
+	It("Healthcheck failure test", func() {
+		mHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
```

Comment:
please use httptest

Reply:
I need to create handler as WithHealthMiddleware requires it.
In anycase there is no need for httptest here, i don't see anything it gives us